server.go

```diff
@@ -1707,6 +1715,20 @@ func (s *Server) handleStream(t transport.ServerTransport, stream *transport.Str
 		}
 	}
 
+	md, _ := metadata.FromIncomingContext(ctx)
+	for _, sh := range s.opts.statsHandlers {
+		ctx = sh.TagRPC(ctx, &stats.RPCTagInfo{FullMethodName: stream.Method()})
+		stream.SetContext(ctx) // To have calls in stream callouts work. Will delete once all stats handler calls come from the gRPC layer.
```

Comment:
You can move this to after the loop so it only happens once.

Reply:
Good point. Done.